### ISSUE TITLE
Project dependencies may have API risk issues

### ISSUE THREAD

**Issue by PyDeps** [PaulDanielML/MuJoCo_RL_UR5#16]
Hi, In **MuJoCo_RL_UR5**, inappropriate dependency versioning constraints can cause risks.

Below are the dependencies and version constraints that the project is using

```
mujoco_py
numpy
simple_pid
termcolor
ikpy
pyquaternion
opencv-python
matplotlib
gym
```

The version constraint **==** will introduce the risk of dependency conflicts because the scope of dependencies is too strict.
The version constraint **No Upper Bound** and **\*** will introduce the risk of the missing API Error because the latest version of the dependencies may remove some APIs.

After further analysis, in this project,
The version constraint of dependency **numpy** can be changed to *>=1.8.0,<=1.23.0rc3*.

The above modification suggestions can reduce the dependency conflicts as much as possible, 
and introduce the latest version as much as possible without calling Error in the projects.

The invocation of the current project includes all the following methods.

<details><summary>The calling methods from the numpy</summary>
 <pre>numpy.linalg.inv
</pre>
</details>
<details><summary>The calling methods from the all methods</summary>
 <pre>numpy.round
agent.optimizer.state_dict
numpy.min
self.viewer.render
self.controller.pixel_2_world
self.policy_net.load_state_dict
self.standardize_rgb
max_value.item
collections.defaultdict
numpy.concatenate
depth_list.append
depth_tensor.detach.clone
self.optimizer.zero_grad
torch.optim.Adam
abs
self.controller.depth_2_meters
device.state.to.self.policy_net.view
torchvision.transforms.Normalize
Grasp_Agent.learn
fig.add_subplot.set_ylabel
str
parameter.numel
pixel.np.round.astype
AverageMeter.update
Modules.MULTIDISCRETE_RESNET.to.eval
self.ee_chain.inverse_kinematics
model.named_parameters
torch.optim.Adam.zero_grad
self.bn3
collections.defaultdict.append
targets.positive_predection.int
data.unsqueeze
collections.deque
numpy.sqrt
self.standardize_depth
matplotlib.pyplot.subplots_adjust
numpy.array.flatten
self.ee_chain.forward_kinematics
gym_grasper.controller.MujocoController.MJ_Controller.move_ee
self.add_marker
self.UP2
targets.sum.item
gym.spaces.MultiDiscrete
agent.env.step
Grasping_Agent_multidiscrete.Grasp_Agent.update_tensorboard
mujoco_py.load_model_from_path
agent.writer.close
kwargs.items
open
max_idx.view.view
self.standardize_depth.detach
conv2d_size_out
Grasping_Module_multidiscrete
grad.view.cpu
pickle.loads
self.policy_net.state_dict
self.render
torch.load.keys
self.last_1000_rewards.append
collections.defaultdict.items
self.model.actuator_id2name
idx_list.append
self.bn2
gym_grasper.controller.MujocoController.MJ_Controller.toss_it_from_the_ellbow
gym.make.reset
self.plot_list.append
torch.nn.Sequential
controller
self.writer.add_scalars
sys.path.insert
math.radians
pyquaternion.Quaternion.random
self.target_net.eval
Grasping_Agent_multidiscrete.Grasp_Agent
abs.dot
torch.optim.Adam.step
gym.utils.EzPickle.__init__
rgb_list.append
Grasping_Agent_multidiscrete.Grasp_Agent.transform_action
format
time.strftime
test
torchvision.transforms.ToTensor
torch.nn.functional.binary_cross_entropy.backward
end_idx.start_idx.batch.action.torch.cat.to
self.model.actuator_id2name.append
Perception_Module
torch.device
self.Grasping_Module_multidiscrete.super.__init__
self.C1
self.RB1
ValueError
torch.optim.Adam.load_state_dict
torchvision.transforms.ColorJitter
simple_pid.PID
self.get_mean_std
self.viewer.add_marker
Modules.ReplayBuffer
gym.envs.mujoco.mujoco_env.MujocoEnv.close
torch.randn_like
self.rotate_wrist_3_joint_to_value
self.env.controller.pixel_2_world
fmtstr.format
prettytable.PrettyTable
numpy.flipud
torch.utils.tensorboard.SummaryWriter.add_scalar
rewards.append
torch.zeros
numpy.fliplr
Transition
pickle.dump
fig.add_subplot.plot
self.controller.set_group_joint_target
self.controller.stay
self.writer.add_scalar
cv2.imwrite
torch.optim.SGD
copy.deepcopy
policy_net
self.BasicBlock.super.__init__
torch.nn.functional.relu
inspect.signature
axis.yaxis.set_label_coords
torch.nn.Linear
matplotlib.pyplot.clf
zip
numpy.linalg.inv
self.C1.size
self.fc1
self.controller.move_group_to_joint_target
self.model.camera_name2id
func
norm_layer
termcolor.colored
self.normal_rgb_no_jitter_no_noise
self.env.model.camera_name2id
matplotlib.pyplot.savefig
max_idx.view.unsqueeze_
conv3x3
agent.policy_net.state_dict
random.randrange
i.self.actuators.self.plot_list.append
func.keys
targets.sum
int
self.ee_chain.inverse_kinematics_frame
range
self.CONV3_FC1.super.__init__
axis.xaxis.set_label_coords
torch.nn.MaxPool2d
numpy.ones
gym.envs.registration.register
data.len.state_batch.policy_net.view.gather
self.close_gripper
self.n_actions_2.MULTIDISCRETE_RESNET.to
self.HEIGHT.self.WIDTH.torch.zeros.to
self.create_group
resnet
isinstance
numpy.max
numpy.tan
state_batch.policy_net.view
self.controller_list.append
self.model.body_name2id
Grasping_Agent_multidiscrete.Grasp_Agent.epsilon_greedy
numpy.std
self.controller.move_ee
agent.transform_observation.to
self.UP1
self.optimizer.step
end_idx.start_idx.batch.state.torch.cat.to
self.C1.view
torch.utils.data.DataLoader
states.append
torch.cuda.is_available
inspect.signature.bind_partial
self.model.body_id2name
self.memory.sample
numpy.reshape
cv2.imshow
simple_Transition
self.model.joint_id2name
print
self.normal_rgb
max_idx.unsqueeze_.cpu
sys.path.append
self.controller.ik
self.controller.open_gripper
targets.size
matplotlib.pyplot.figure
self.controller.grasp
self.writer.add_histogram
positive_threshold.predictions.int
random.seed
tqdm.tqdm
self.bn1
bound_values.arguments.items
AverageMeter.reset
random.sample.append
time.localtime
NotImplementedError
gym.spaces.Discrete
numpy.where
rgb.self.standardize_rgb.float
self.controller.close_gripper
Modules.MULTIDISCRETE_RESNET.to.state_dict
self.ik
self.last_100_loss.append
pathlib.Path
self.conv1
cv2.waitKey
self.conv3
self.create_camera_data
grad.view.cpu.numpy
self.reset
torch.save
observation.astype
torch.tensor
next_state_batch.self.target_net.max
self.relu
setuptools.setup
end_idx.start_idx.batch.reward.torch.cat.to.float
ljust
gym_grasper.controller.MujocoController.MJ_Controller.show_model_info
self.writer.add_graph
data.unsqueeze.to
typeassert
env.action_space.sample
time.time
self.C1.squeeze_
self.policy_net.parameters
file.read
main
enumerate
Modules.Transition
torch.utils.tensorboard.SummaryWriter
prettytable.PrettyTable.add_row
collections.namedtuple
repr
self.conv2
ikpy.chain.Chain.from_urdf_file
torch.nn.Conv2d
targets.int
torch.cuda.manual_seed
math.exp
numpy.random.seed
negative_threshold.predictions.int
negative_positions.correct_prediction_2.sum.item
self.move_and_grasp
Grasping_Agent_multidiscrete.Grasp_Agent.transform_observation
fig.add_subplot.set_xlabel
agent.env.reset
Modules.MULTIDISCRETE_RESNET.to.load_state_dict
Modules.MULTIDISCRETE_RESNET
Grasp_Agent.update_tensorboard
gym.make
self.sim.step
ee_position.copy
depth.torch.tensor.float
self.create_joint_angle_plot
self.sim.data.get_joint_qpos
pyquaternion.Quaternion
gym_grasper.controller.MujocoController.MJ_Controller
numpy.array
cv2.cvtColor
self.policy_net
MAX_POSSIBLE_SAMPLES.state_batch.self.policy_net.view.gather
torchvision.transforms.Lambda
next_state_batch.self.target_net.max.unsqueeze
self.transform_observation
torch.manual_seed
gym.make.step
self.fill_plot_list
agent.env.close
torch.nn.BatchNorm2d
self.memory.append
random.sample
numpy.mean
torch.utils.data.random_split
matplotlib.pyplot.figure.add_subplot
self.target_net.load_state_dict
max
actions.append
agent.optimizer.zero_grad
func.items
torch.nn.ReLU
fig.add_subplot.axhline
gym.make.print_info
rgb.self.normal_rgb.float
grasping_dataset.Grasping_Dataset
data.to
targets.negative_predection.int
torch.nn.functional.binary_cross_entropy
self.last_1000_actions.append
Modules.simple_Transition
Grasp_Agent
targets.correct_prediction.sum.item
gym_grasper.controller.MujocoController.MJ_Controller.stay
self.MP2
end_idx.start_idx.batch.next_state.torch.cat.to
Modules.MULTIDISCRETE_RESNET.to.parameters
self.actuators.append
self.sigmoid
self.create_lists
self.Grasping_Module.super.__init__
type
numpy.argmax
Grasp_Agent.transform_action
self.optimizer.load_state_dict
binary_accuracy
self.RB2
list
torch.cat
os.path.realpath
rgb.self.normal_rgb_no_jitter_no_noise.float
count_parameters
len
torchvision.transforms.Compose
self.MP1
time.perf_counter
mujoco_py.MjViewer
self.get_observation
data.unsqueeze.to.float
self.sim.render
random.random
numpy.zeros
eval_on_test_set
self.Perception_Module.super.__init__
mujoco_py.MjSim
numpy.expand_dims
coordinates_2.np.round.np.ndarray.coordinates_2.isinstance.ljust
torch.nn.Sigmoid
torch.nn.UpsamplingBilinear2d
inspect.signature.bind
self.set_state
self.plot_list.keys
next_state_batch.self.target_net.max.unsqueeze.detach
targets.correct_prediction.sum
Grasping_Module
AverageMeter
agent.epsilon_greedy.item
gym.make.close
torch.nn.functional.binary_cross_entropy.item
MULTIDISCRETE_RESNET
Grasp_Agent.epsilon_greedy
negative_positions.correct_prediction_2.sum
gym_grasper.controller.MujocoController.MJ_Controller.grasp
agent.memory.push
self.target_net
torch.optim.Adam.state_dict
j.self.actuators
torch.Tensor
join
cv2.destroyAllWindows
fig.add_subplot.set_title
self.RB3
Grasping_Agent_multidiscrete.Grasp_Agent.learn
gym.envs.mujoco.mujoco_env.MujocoEnv.__init__
device.state.to.self.policy_net.view.max
torch.load
numpy.random.normal
TypeError
super
state_batch.self.policy_net.view
Modules.MULTIDISCRETE_RESNET.to
self.controller.get_image_data
depth_tensor.detach.clone.unsqueeze_
grad.view
torchvision.transforms.ToPILImage
self.move_group_to_joint_target
self.groups.keys
self.model.get_joint_qpos_addr
Grasp_Agent.transform_observation
end_idx.start_idx.batch.reward.torch.cat.to
self.model.camera_id2name
torch.no_grad
functools.wraps
numpy.random.uniform
BasicBlock
</pre>
</details>

@developer
Could please help me check this issue?
May I pull a request to fix it?
Thank you very much.
